tachyon/zk/plonk/halo2/proof_unittest.cc

```diff
@@ -59,23 +61,30 @@ TEST(ProofTest, JsonValueConverter) {
   expected_proof.vanishing_random_poly_commitment = Commitment::Random();
   expected_proof.y = F::Random();
   expected_proof.vanishing_h_poly_commitments =
-      base::CreateVector(5, Commitment::Random());
+      base::CreateVector(5, []() { return Commitment::Random(); });
   expected_proof.x = F::Random();
   expected_proof.instance_evals_vec =
       CreateRandomElementsVec<F>(num_circuits_, num_elements_);
   expected_proof.advice_evals_vec =
       CreateRandomElementsVec<F>(num_circuits_, num_elements_);
-  expected_proof.fixed_evals = base::CreateVector(num_circuits_, F::Random());
+  expected_proof.fixed_evals =
+      base::CreateVector(num_circuits_, []() { return F::Random(); });
   expected_proof.vanishing_random_eval = F::Random();
   expected_proof.common_permutation_evals =
-      base::CreateVector(num_circuits_, F::Random());
+      base::CreateVector(num_circuits_, []() { return F::Random(); });
   expected_proof.permutation_product_evals_vec =
       CreateRandomElementsVec<F>(num_circuits_, num_elements_);
   expected_proof.permutation_product_next_evals_vec =
       CreateRandomElementsVec<F>(num_circuits_, num_elements_);
-  expected_proof.permutation_product_last_evals_vec = {
-      base::CreateVector(5, std::optional<F>(F::Random())),
-      base::CreateVector(5, std::optional<F>())};
+  expected_proof.permutation_product_last_evals_vec =
+      std::vector<std::vector<std::optional<F>>>(
+          num_circuits_, base::CreateVector(num_elements_, [](size_t i) {
+            if (i % 2 == 0) {
+              return std::optional<F>(F::Random());
+            } else {
+              return std::optional<F>();
+            }
+          }));
```

Comment:
Please note that this vector should contain 1 None at the front, and the remaining 9 elements should be some random values. However, since it's just for testing JSON conversion, I'll ignore this requirement for now.